pkgs/http_profile/lib/src/http_profile_request_data.dart

```diff
@@ -46,20 +62,43 @@ final class HttpProfileRequestData {
   /// This information is meant to be used for debugging.
   ///
   /// It can contain any arbitrary data as long as the values are of type
-  /// [String] or [int]. For example:
-  /// { 'localPort': 1285, 'remotePort': 443, 'connectionPoolId': '21x23' }
-  set connectionInfo(Map<String, dynamic /*String|int*/ > value) {
+  /// [String] or [int].
+  ///
+  /// This field can only be modified by assigning a Map to it. That is:
+  /// ```dart
+  /// // Valid
+  /// profile?.requestData.connectionInfo = {
+  ///   'localPort': 1285,
+  ///   'remotePort': 443,
+  ///   'connectionPoolId': '21x23',
+  /// };
+  ///
+  /// // Invalid
+  /// profile?.requestData.connectionInfo?['localPort'] = 1285;
+  /// ```
+  set connectionInfo(Map<String, dynamic /*String|int*/ >? value) {
```

Comment:
I had to give the new getter a nullable return type, because `_requestData` won't have a `'connectionInfo'` entry until the setter is called. The setter's parameter type needs to be a supertype of the getter's return type, so my options were to make it nullable as well, or give the getter and setter different names. I chose to make the setter's parameter type nullable. Does that seem ok?